src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/SkipExecution.kt

```diff
@@ -0,0 +1,58 @@
+package com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.util.OpenForTesting
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoAction
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoRequest
+import org.elasticsearch.action.admin.cluster.node.info.NodesInfoResponse
+import org.elasticsearch.action.admin.cluster.node.info.PluginsAndModules
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.ClusterChangedEvent
+import org.elasticsearch.cluster.ClusterStateListener
+import org.elasticsearch.cluster.service.ClusterService
+
+@OpenForTesting
+class SkipExecution(
+    private val client: Client,
+    private val clusterService: ClusterService
+) : ClusterStateListener {
+    private val logger = LogManager.getLogger(javaClass)
+
+    final var flag: Boolean = false
```

Comment:
Should we set this to volatile?

Reply:
makes sense, there could be multiple node change event happened at same time